### PR TITLE
update ctc_beam_search_decoder design doc

### PR DESCRIPTION
**Pull request by kuke**
update design doc for beam search with CTC and LM  #2230

### CODE REVIEW

File: doc/design/speech/README.MD [hunk 1]
```diff
@@ -140,7 +140,17 @@ TODO by Assignees
...
 ### Beam Search with CTC and LM
 
-TODO by Assignees
+<div align="center">
+<img src="image/beam_search.png" width=400><br/>
+Figure 2. Algorithm for Beam Search Decoder.
+</div>
+
+- The **Beam Search Decoder** for DS2 CTC-trained network follows the similar approach in \[[3](#references)\] with a modification for the ambiguous part, as shown in Figure 2.
+- An **external defined scorer** would be passed into the decoder to evaluate a candidate prefix during decoding whenever a space character appended.
```
Comment (xinghai-sun) [inline doc/design/speech/README.MD]: remove "defined"

File: doc/design/speech/README.MD [hunk 2]
```diff
@@ -140,7 +140,17 @@ TODO by Assignees
 
 ### Beam Search with CTC and LM
 
-TODO by Assignees
+<div align="center">
+<img src="image/beam_search.png" width=400><br/>
+Figure 2. Algorithm for Beam Search Decoder.
+</div>
+
+- The **Beam Search Decoder** for DS2 CTC-trained network follows the similar approach in \[[3](#references)\] with a modification for the ambiguous part, as shown in Figure 2.
```
Comment (xinghai-sun) [inline doc/design/speech/README.MD]: What is the ambiguous part and what is the modification? Could you please add more details?

File: doc/design/speech/README.MD [hunk 3]
```diff
@@ -140,7 +140,17 @@ TODO by Assignees
...
+- An **external defined scorer** would be passed into the decoder to evaluate a candidate prefix during decoding whenever a space character appended.
+- Such scorer is a unified class, may consisting of language model, word count or any customed evaluators.
```
Comment (xinghai-sun) [inline doc/design/speech/README.MD]: 1. --> any other custom scorers.
2. What do you mean by "a unified class"?
3. Remove "may".

File: doc/design/speech/README.MD [hunk 4]
```diff
@@ -140,7 +140,17 @@ TODO by Assignees
...
+- The **language model** is built from Task 5, with a parameter should be carefully tuned to achieve minimum WER/CER (c.f. Task 7)
```
Comment (xinghai-sun) [inline doc/design/speech/README.MD]: a parameters --> parameters

File: doc/design/speech/README.MD (hunk 2 above)
Comment (kuke) [reply doc/design/speech/README.MD]: Done

File: doc/design/speech/README.MD (hunk 1 above)
Comment (kuke) [reply doc/design/speech/README.MD]: Done

File: doc/design/speech/README.MD (hunk 3 above)
Comment (kuke) [reply doc/design/speech/README.MD]: Done and modified.

File: doc/design/speech/README.MD (hunk 4 above)
Comment (kuke) [reply doc/design/speech/README.MD]: Done

**Pull request merged**